motor/discovery/azure/mql_assets.go

```diff
@@ -0,0 +1,226 @@
+	}
+	m, err := NewMQLAssetsDiscovery(provider)
+	if err != nil {
+		return nil, err
+	}
+	// TODO: make this work by having the ip addresses available via mql
+	// if tc.IncludesOneOfDiscoveryTarget(common.DiscoveryAll, DiscoveryInstances) {
+	// 	instances, err := computeInstances(m, provider.SubscriptionID(), pCfg)
+	// 	if err != nil {
+	// 		return nil, err
+	// 	}
+	// 	assets = append(assets, instances...)
+	// }
+	if tc.IncludesOneOfDiscoveryTarget(common.DiscoveryAll, DiscoverySqlServers) {
+		servers, err := computeSqlServers(m, provider.SubscriptionID(), pCfg)
+		if err != nil {
+			return nil, err
+		}
+		assets = append(assets, servers...)
+	}
+	if tc.IncludesOneOfDiscoveryTarget(common.DiscoveryAll, DiscoveryPostgresServers) {
+		servers, err := computePostgresqlServers(m, provider.SubscriptionID(), pCfg)
+		if err != nil {
+			return nil, err
+		}
+		assets = append(assets, servers...)
+	}
+	if tc.IncludesOneOfDiscoveryTarget(common.DiscoveryAll, DiscoveryMySqlServers) {
+		servers, err := computeMySqlServers(m, provider.SubscriptionID(), pCfg)
+		if err != nil {
+			return nil, err
+		}
+		assets = append(assets, servers...)
+	}
+	if tc.IncludesOneOfDiscoveryTarget(common.DiscoveryAll, DiscoveryMariaDbServers) {
+		servers, err := computeMariaDbServers(m, provider.SubscriptionID(), pCfg)
+		if err != nil {
+			return nil, err
+		}
+		assets = append(assets, servers...)
+	}
+	if tc.IncludesOneOfDiscoveryTarget(common.DiscoveryAll, DiscoveryStorageAccounts) {
+		accounts, err := computeStorageAccounts(m, provider.SubscriptionID(), pCfg)
+		if err != nil {
+			return nil, err
+		}
+		assets = append(assets, accounts...)
+	}
+	if tc.IncludesOneOfDiscoveryTarget(common.DiscoveryAll, DiscoveryStorageContainers) {
+		containers, err := computeStorageAccountContainers(m, provider.SubscriptionID(), pCfg)
+		if err != nil {
+			return nil, err
+		}
+		assets = append(assets, containers...)
+	}
+	if tc.IncludesOneOfDiscoveryTarget(common.DiscoveryAll, DiscoveryKeyVaults) {
+		vaults, err := computeKeyVaultsVaults(m, provider.SubscriptionID(), pCfg)
+		if err != nil {
+			return nil, err
+		}
+		assets = append(assets, vaults...)
+	}
+	if tc.IncludesOneOfDiscoveryTarget(common.DiscoveryAll, DiscoverySecurityGroups) {
+		securityGroups, err := computeNetworkSecurityGroups(m, provider.SubscriptionID(), pCfg)
+		if err != nil {
+			return nil, err
+		}
+		assets = append(assets, securityGroups...)
+	}
+
+	return assets, nil
+}
+
+type mqlObject struct {
+	name        string
+	labels      map[string]string
+	azureObject azureObject
+}
+
+type azureObject struct {
+	subscription string
+	id           string
+	region       string
+	service      string
+	objectType   string
+}
+
+type azureObjectPlatformInfo struct {
+	title    string
+	platform string
+}
+
+func AzureObjectPlatformId(azureObject azureObject) string {
+	// the azure resources have an unique id throughout the whole subscription
```

Comment:
What about multiple subscriptions? Should we add the subscription to the ID?

Reply:
I added an example id and updated that comment, the azure id also has the subscription id as part of it so that guarantees uniqueness throughout multiple subs too